sentinel/sentinel.go

```diff
@@ -122,6 +126,16 @@ func NewClient(
 ) (
 	*Client, error,
 ) {
+	return NewClientCustom(network, address, poolSize, redis.Dial, names...)
+}
+
+// NewClientCustom the same as NewClient except you can specify a DialFunc that will
```

Comment:
If you could change this to:

```go
// NewClientCustom is the same as NewClient, except it takes in a DialFunc which                                                                                                                                                             
// will be used to create all new connections to the master instances. This can  
// be used to implement authentication, custom timeouts, etc...
```

Sorry, I'm picky about my comments :P